### DISCUSSION TITLE
tutorials/assets_pipeline/importing_3d_scenes/available_formats

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#236] [User-contributed notes]
# tutorials/assets_pipeline/importing_3d_scenes/available_formats

When dealing with 3D assets, Godot has a flexible and configurable importer. Godot works with scenes. This means that the entire scene being worked on in your favorite 3D modeling software will be ...

https://docs.godotengine.org/en/latest/tutorials/assets_pipeline/importing_3d_scenes/available_formats.html

**Comment by yourboirusty** [comment]


**Reply by Calinou** [reply]
Import hints are documented here: https://docs.godotengine.org/en/latest/tutorials/assets_pipeline/importing_3d_scenes/node_type_customization.html#doc-importing-3d-scenes-node-type-customization

**Comment by stephen-berry** [comment]
Were there discussions for adding the .blend for Blender 4.x? Assuming we use the newer blender versions, there's no option in the settings for a blender path, and the docs here mention the path is titled "-> Blender 3 Path"

**Reply by rteijeiro** [reply]
Go to Editor -> Editor Settings... -> FileSystem -> Import

There you should be able to see the Blender Path option on the right.

**Reply by stephen-berry** [reply]
Thanks! I was in project settings :/

**Comment by emanvidmaker** [comment]
Note that

To access custom properties from blender inside of Godot, you need to use get_meta(&"extras") on the node that represents that object, and it should return a dictionary that has all the custom properties you set in blender

**Comment by BlueG** [comment]
**Recommendation**: Always assign a root node type of `Node3D` (or a descendant) and give the root node a name when importing Blender or GLTF scenes (I haven't tried other formats).

I had trouble re-importing either Blender or GLTF files. The original import worked, but when I placed them in a scene, they were given the suffix `2` on the first instance (there was no name conflict without the suffix). And when I went to reimport them, instances in the scene were not updated. Assigning the root node type fixed the issue where a reimport wasn't updating scene instances and assigning the root a name fixed the suffix issue.

I'm documenting this here for anyone else trying to import these file types. I'll try to create a GitHub issue with more details later.
